### ISSUE TITLE
Order of JS events might be wrong

### ISSUE THREAD

**Issue by abitdodgy** [refile/refile#326]
On master, it seems that `upload:complete` is being fired before `upload:success` or `upload:failure`. Is this the right behaviour, or a bug? I think it makes more sense for `complete` to fire after. I've created [a pull request][1] to address this issue.

[1]: https://github.com/refile/refile/pull/325

**Comment by trentclowater** [comment]
I believe this is the right behaviour (many other javascript libraries use the same order). The server returns `success` or `error`, and then the process is `complete`. It can't be `complete` if it is unknown whether the upload was a `success` or if there was an `error`.

For reference, here is the jQuery ajax events page: http://api.jquery.com/Ajax_Events/ 

A few other ajax/upload libraries I have looked at also use the same order/terminology.

**Comment by abitdodgy** [comment]
@trentclowater so this is a bug then? Because as it stands complete is being fired *before* `success` or `failure`, not after.

**Comment by trentclowater** [comment]
Sorry, read your comment wrong. If `complete` is being fired before `success` or `failure`, then that does seem to be a different order than any other javascript libraries I have ever used.

Probably also explains why when I was first implementing direct uploads in my application, I was confused why it wasn't working. But then I switched the event and it worked, and I had to move on to other things...
